prometheus/go_collector.go

```diff
@@ -9,27 +10,245 @@ import (
 type goCollector struct {
 	goroutines Gauge
 	gcDesc     *Desc
+
+	// memstats object to reuse
```

Comment:
Re-using will create a race. Doc for Collector.Collect(): "This method may be called concurrently and must therefore be implemented in a concurrency safe way. "